### DISCUSSION TITLE
How to play multiple sounds at the same time in app?

### DISCUSSION THREAD

**Discussion by gerard16** [bluefireteam/audioplayers#1386] [General]
I have the below code where the function clickSound just plays a simple button click sound audio without any loop and the function loopSound plays music in a loop.However I am unable to play both at the same time in case the music is playing and I want to click a button for example.

I assume that I will have to create two instances and will not be able to do that with a singleton?
Or am I doing something wrong?

```
import 'package:audioplayers/audioplayers.dart';
import 'package:biblequiz/services/preferences.dart';

class AudioService {
  AudioPlayer player = AudioPlayer();

  static final AudioService instance = AudioService._();

  AudioService._() {
    this.player.setVolume(1.0);
  }

  void clickSound() async {
    await player.play(AssetSource('sounds/click.mp3'));      
  }

  void loopSound() async {
      player.setReleaseMode(ReleaseMode.loop);
      await player.play(AssetSource('sounds/music.wav'));
  }

  void stopLoop() async {
    await player.stop();
  }
}
```

**Comment by Gustl22** [comment]
Yes you would need multiple `AudioPlayer` objects (or a list of `AudioPlayer`s) within your AudioService instance.
You may have a look at `AudioPool` of the Flame package:
https://github.com/flame-engine/flame/blob/main/packages/flame_audio/lib/audio_pool.dart

Edit: I think you created the same discussion twice: see #1387
